src/main/java/interface_adapters/tui/controllers/RecipeNoteOperation.java

```diff
@@ -0,0 +1,177 @@
+            new NoteCreator(),
+            new NoteViewer(),
+            new NoteEditor(),
+            new NoteDeleter()
+    );
+
+    public RecipeNoteOperation(TextualReader reader, RecipeNoteTaker recipeNoteTaker, RecipeManager recipeManager) {
+        this.reader = reader;
+        this.recipeNoteTaker = recipeNoteTaker;
+        this.recipeManager = recipeManager;
+    }
+
+    @Override
+    public void run() {
+        reader.chooseOperation(operations, "Recipe Note Taker");
+    }
+
+    @Override
+    public String getDescription() {
+        return "Create, list, edit, and delete notes";
+    }
+
+    @Override
+    public String getCode() {
+        return "recipe note taker";
+    }
+
+    private class NoteCreator implements TextualOperation {
+
+        @Override
+        public String getCode() {
+            return "create note";
+        }
+
+        @Override
+        public String getDescription() {
+            return "Create a new note";
+        }
+
+        @Override
+        public void run() {
+
+            List<Recipe> recipes = recipeManager.getAllRecipes();
+            Recipe recipe = reader.chooseFromList(recipes, "Recipe to Add Note");
+
+            if(recipe == null){
+                return;
+            }
+
+            String noteText = reader.getInput("Note Text:");
+
+            Note note = new Note(noteText, recipe);
+
+            recipeNoteTaker.createNote(note);
+
+            Colour.info("Created note for recipe %s: %s", recipe.getName(), noteText);
+        }
+
+    }
+
+    private class NoteViewer implements TextualOperation {
+        @Override
+        public String getCode() {
+            return "view note";
+        }
+
+        @Override
+        public String getDescription() {
+            return "View a saved note";
+        }
+
+        @Override
+        public void run() {
+
+            List<Note> notes = recipeNoteTaker.getAllNotes();
+
+            Note note = reader.chooseFromList(notes, "All Notes");
+
+            if (note != null) {
+                System.out.println(note);
+            }
+        }
+
+    }
+
+    private class NoteEditor implements TextualOperation {
+
+        @Override
+        public String getCode() {
+            return "edit note";
+        }
+
+        @Override
+        public String getDescription() {
+            return "Edit an existing note";
+        }
+
+        @Override
+        public void run() {
+            //List<Recipe> recipes = recipeManager.getAllRecipes();
+            //Recipe recipe = reader.chooseFromList(recipes, "Recipe to select");
+
+            List<Note> notes = recipeNoteTaker.getAllNotes();
+            Note note = reader.chooseFromList(notes, "Note to Edit");
+
+            if (note == null) {
+                return;
+            }
+
+            //reader.chooseOperation(operations, "Choose Edit");
+
+            //String title = reader.getInput("Recipe Title:");
```

Comment:
Any code that isn't being used in the final PR can be deleted.

Reply:
ok